python/pyarrow/tests/test_udf.py

```diff
@@ -0,0 +1,483 @@
+                                },
+                                pa.int64())
+    return ternary_function, func_name
+
+
+@pytest.fixture(scope="session")
+def varargs_func_fixture():
+    def varargs_function(ctx, *values):
+        base_val = values[:2]
+        res = pc.call_function("add", base_val)
+        for other_val in values[2:]:
+            res = pc.call_function("add", [res, other_val])
+        return res
+    func_name = "z=ax+by+c"
+    varargs_doc = {"summary": "z=ax+by+c",
+                   "description": "find z from z = ax + by + c"
+                   }
+    pc.register_scalar_function(varargs_function,
+                                func_name,
+                                varargs_doc,
+                                {
+                                    "array1": pa.int64(),
+                                    "array2": pa.int64(),
+                                    "array3": pa.int64(),
+                                    "array4": pa.int64(),
+                                    "array5": pa.int64(),
+                                },
+                                pa.int64())
+    return varargs_function, func_name
+
+
+@pytest.fixture(scope="session")
+def random_with_udf_ctx_func_fixture():
+    def random_with_udf_ctx(context, one, two):
+        proxy_pool = pa.proxy_memory_pool(context.memory_pool)
+        ans = pc.add(one, two, memory_pool=proxy_pool)
+        res = pa.array([ans.as_py()], memory_pool=proxy_pool)
+        return res
+    in_types = {"one": pa.int64(),
+                "two": pa.int64(),
+                }
+    func_doc = {
+        "summary": "test udf context",
+        "description": "udf context test"
+    }
+    func_name = "test_udf_context"
+    pc.register_scalar_function(random_with_udf_ctx,
+                                func_name, func_doc,
+                                in_types,
+                                pa.int64())
+    return random_with_udf_ctx, func_name
+
+
+def const_return(ctx, scalar):
+    return 42
+
+
+@pytest.fixture(scope="session")
+def output_check_func_fixture():
+    def output_check(ctx, array):
+        ar = pc.call_function("add", [array, 1])
+        ar = ar.cast(pa.int32())
+        return ar
+    func_name = "test_output_value"
+    in_types = {"array": pa.int64()}
+    out_type = pa.int64()
+    doc = {
+        "summary": "add function scalar",
+        "description": "add function"
+    }
+    pc.register_scalar_function(output_check, func_name, doc,
+                                in_types, out_type)
+    return output_check, func_name
+
+
+@pytest.fixture(scope="session")
+def nullary_check_func_fixture():
+    # this needs to return array values
+    def nullary_check(ctx):
+        rand_vals = []
+        print("batch_length: ", ctx.batch_length)
+        for _ in range(ctx.batch_length):
+            random.seed(10)
+            rand_vals.append(random.randint(0, 10))
```

Comment:
Hmm, do you realize this always appends the same value?
```python
>>> random.seed(10)
>>> random.randint(0, 10)
9
>>> random.seed(10)
>>> random.randint(0, 10)
9
```

Reply:
Also, you can just use Numpy's random generation facilities: https://numpy.org/doc/stable/reference/random/generator.html

Reply:
Yes, the expectation is to keep it the same. I did that to assert the values easily.

Reply:
Yes, I kind of deviated from the objective. You're correct.